### ISSUE TITLE
NeedMoreInput exception

### ISSUE THREAD

**Issue by nirlachman-lp** [tidwall/SwiftWebSocket#43]
Hi,
I integrated the lib in my project and just simple tried to create a socket and open (thats it).
I keep receiving NeedMoreInput exception and I have no other error.
I tried for simplicity to use this url:
ws://echo.websocket.org

Throwing line is 1115 in readResponse()

What can be the cause?
Thanks!

**Comment by tidwall** [comment]
Hi @nirlachman-liveperson. I'm sorry to hear about the problems you are having with the library. Hopefully I can help. What's your development environment and do you have some example code to reproduce the error?  Thanks!

**Comment by nirlachman-lp** [comment]
I'm using the latest XCode (7.3.1) with the latest Swift SDK.
In my main VC in viewDidLoad I just do:
let ws = NLWebSocket("ws://echo.websocket.org")

Event when i don't call open(), its crashing.
The weird thing is when i'm trying to just run it in a different sample app, it works.
Is there anything in the build settings which can cause this issue? or any other configuration?
Just to mention: I linked libz.tbd

Thanks

**Comment by nirlachman-lp** [comment]
Hey,
Another update on the issue:
When I run the app in RELEASE mode it runs fine, when moving to DEBUG it keeps crashing.
Hope it helps

**Comment by tidwall** [comment]
I just tried `let ws = WebSocket("ws://echo.websocket.org")` in `viewDidLoad` under DEBUG and RELEASE, and did not have any crashes or errors.

I also tested the [example](https://github.com/tidwall/SwiftWebSocket#example) code from the README and it's working as expected.

What version of SwiftWebSocket are you using? 

Also `NLWebSocket` isn't a function in the library. Does it belong to another library or is it custom to your project?
